src/adapter/src/catalog/storage.rs

```diff
@@ -1731,46 +1731,79 @@ where
 pub async fn initialize_stash(stash: &mut Stash) -> Result<(), Error> {
     async fn add_batch<'tx, K, V>(
         tx: &'tx mz_stash::Transaction<'tx>,
-        batches: &mut Vec<AppendBatch>,
         typed: &TypedCollection<K, V>,
-    ) -> Result<(), StashError>
+    ) -> Result<Option<AppendBatch>, StashError>
     where
         K: mz_stash::Data,
         V: mz_stash::Data,
     {
         let collection = tx.collection::<K, V>(typed.name()).await?;
         let upper = tx.upper(collection.id).await?;
         if upper.elements() == [mz_stash::Timestamp::MIN] {
-            let batch = collection.make_batch_lower(upper)?;
-            batches.push(batch);
+            Ok(Some(collection.make_batch_lower(upper)?))
+        } else {
+            Ok(None)
         }
-        Ok(())
     }
 
     stash
         .with_transaction(move |tx| {
             Box::pin(async move {
-                let mut batches = Vec::new();
-                add_batch(&tx, &mut batches, &COLLECTION_CONFIG).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_SETTING).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_ID_ALLOC).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_SYSTEM_GID_MAPPING).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_CLUSTERS).await?;
-                add_batch(
-                    &tx,
-                    &mut batches,
-                    &COLLECTION_CLUSTER_INTROSPECTION_SOURCE_INDEX,
-                )
-                .await?;
-                add_batch(&tx, &mut batches, &COLLECTION_CLUSTER_REPLICAS).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_DATABASE).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_SCHEMA).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_ITEM).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_ROLE).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_TIMESTAMP).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_SYSTEM_CONFIGURATION).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_AUDIT_LOG).await?;
-                add_batch(&tx, &mut batches, &COLLECTION_STORAGE_USAGE).await?;
+                // Query all collections in parallel. Makes for triplicated
+                // names, but runs quick.
+                let (
+                    config,
+                    setting,
+                    id_alloc,
+                    system_gid_mapping,
+                    clusters,
+                    cluster_introspection,
+                    cluster_replicas,
+                    database,
+                    schema,
+                    item,
+                    role,
+                    timestamp,
+                    system_configuration,
+                    audit_log,
+                    storage_usage,
+                ) = futures::try_join!(
```

Comment:
@guswynn Is there a better way to do this? Can't use join_all because the futures aren't all the same type due to the KV bounds.